fabric-ca-client/lib/FabricCAServices.js

```diff
@@ -41,18 +40,10 @@ const FabricCAServices = class extends BaseClient {
 	 *
 	 * @param {string | object} url The endpoint URL for Fabric CA services of the form: "http://host:port" or "https://host:port"
 	 When this parameter is an object then it must include the parameters listed as key value pairs.
-	 * @param {TLSOptions} tlsOptions The TLS settings to use when the Fabric CA services endpoint uses "https"
-	 * @param {string} caName The optional name of the CA. Fabric-ca servers support multiple Certificate Authorities from
+	 * @param {TLSOptions} [tlsOptions] The TLS settings to use when the Fabric CA services endpoint uses "https"
+	 * @param {string} [caName] The optional name of the CA. Fabric-ca servers support multiple Certificate Authorities from
 	 *  a single server. If omitted or null or an empty string, then the default CA is the target of requests
-	 * @param {CryptoSuite} cryptoSuite The optional cryptoSuite instance to be used if options other than defaults are needed.
-	 * If not specified, an instance of {@link CryptoSuite} will be constructed based on the current configuration settings:
-	 * <br> - crypto-hsm: use an implementation for Hardware Security Module (if set to true) or software-based key management (if set to false)
-	 * <br> - crypto-keysize: security level, or key size, to use with the digital signature public key algorithm. Currently ECDSA
-	 *  is supported and the valid key sizes are 256 and 384
-	 * <br> - crypto-hash-algo: hashing algorithm
-	 * <br> - key-value-store: some CryptoSuite implementation requires a key store to persist private keys. A {@link CryptoKeyStore}
-	 *  is provided for this purpose, which can be used on top of any implementation of the {@link KeyValueStore} interface,
-	 *  such as a file-based store or a database-based one. The specific implementation is determined by the value of this configuration setting.
+	 * @param {CryptoSuite|ICryptoSuite} [cryptoSuite] The cryptoSuite instance to be used.
```

Comment:
This JSDoc shows `cryptoSuite` as optional but the code (and tests) now make it mandatory. We need to be careful about making breaking changes to existing public APIs. Is it practical (and does it make sense) to keep this as optional with a default value if not specified?

Reply:
And further more,  is it good to accept two fashion of parameter input? people could provide either a string or an object to 1st parameter, And the handling process diverge from its type

I think we should only accept single fashion. Then both document and type definition could be clear.